src/rules/prefer-destructured-store-props.ts

```diff
@@ -0,0 +1,67 @@
+import type { TSESTree } from "@typescript-eslint/types"
+import { createRule } from "../utils"
+
+export default createRule("prefer-destructured-store-props", {
+  meta: {
+    docs: {
+      description: "",
+      category: "Best Practices",
+      recommended: false,
+    },
+    hasSuggestions: true,
+    schema: [],
+    messages: {
+      useDestructuring: `Destructure {{prop}} from store {{store}} for better change tracking & fewer redraws`,
+      fixUseDestructuring: `Using destructuring like $: ({ {{prop}} } = {{store}}); will run faster`,
+    },
+    type: "suggestion",
+  },
+  create(context) {
+    return {
+      // {$foo.bar + baz}
+      // should be
+      // $: ({ bar } = $foo);
+      // {bar + baz}
+      [`MemberExpression[object.name=/$/][property.type="Identifier"]`](
```

Comment:
`/$/` matches the end, so I think it matches all strings. We use `store.startsWith("$")` to check for it, so I think we don't need this selector.

Reply:
Yeah, that's been bugging me and I went back to esquery and I think it should actually be `/^\$/` which means we can remove the check entirely.